cli/src/lockfiles/python.rs

```diff
@@ -88,6 +90,60 @@ impl Parseable for PipFile {
     }
 }
 
+impl Parseable for Poetry {
+    fn new(filename: &Path) -> Result<Self, io::Error>
+    where
+        Self: Sized,
+    {
+        Ok(Poetry(fs::read_to_string(filename)?))
+    }
+
+    /// Parses `poetry.lock` files into a vec of packages
+    fn parse(&self) -> ParseResult {
+        let mut lock: PoetryLock = toml::from_str(&self.0)?;
+
+        // Warn if the version of this lockfile might not be supported.
+        if !lock.metadata.lock_version.starts_with("1.") {
+            log::warn!(
+                "Expected poetry lockfile version ^1.0.0, found {}. \
+                Attempting to continue, but results might be inaccurate.",
+                lock.metadata.lock_version
+            );
+        }
+
+        Ok(lock.packages.drain(..).map(Package::into).collect())
```

Comment:
I think this would be more readable using `PackageDescriptor::from`:

```suggestion
        Ok(lock.packages.drain(..).map(PackageDescriptor::from).collect())
```

I'm guessing the line ended up like this because the natural way to write it is `.map(|p| p.into())`, but clippy doesn't like that. This is one place where I think clippy is wrong....

Reply:
The line ended up that way because it's shorter. I had it as `PackageDescription::from` originally.

Though I do agree that it's more descriptive to show the destination rather than the source type. So it's probably easier for readers to go for the longer solution.